cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/openai/OpenAIPrompt.scala

```diff
@@ -136,10 +192,20 @@ class OpenAIPrompt(override val uid: String) extends Transformer
     }
   }
 
-  override def transformSchema(schema: StructType): StructType =
-    openAICompletion
-      .transformSchema(schema)
-      .add(getPostProcessing, getParser.outputSchema)
+  override def transformSchema(schema: StructType): StructType = {
+    openAICompletion match {
+      case chatCompletion: OpenAIChatCompletion =>
+        chatCompletion.setMessagesCol("messages")
+        chatCompletion
+          .transformSchema(schema)
+          .add(getPostProcessing, getParser.outputSchema)
+      case completion: OpenAICompletion =>
+        completion
+          .transformSchema(schema)
+          .add(getPostProcessing, getParser.outputSchema)
+    }
+
+  }
```

Comment:
here we just want to say how the schema will transform, if its easier to just say how the schema will look we can do that as opoosed to using the transformer itself to figure it out

Reply:
using getMessagesCol now to set the name, as we discussed